daemon/start.go

```diff
@@ -197,4 +198,10 @@ func (daemon *Daemon) Cleanup(container *container.Container) {
 		}
 	}
 	container.CancelAttachContext()
+
+	// if containers AutoRemove flag is set, remove it after clean up
+	if container.HostConfig.AutoRemove {
+		// If containers lock is not released, goroutine will guarantee no block
+		go daemon.ContainerRm(container.ID, &types.ContainerRmConfig{ForceRemove: true, RemoveVolume: true})
```

Comment:
I think I would lean towards not doing this in a goroutine.
If the mutex is locked we likely have bugs elsewhere (which we know we do, of course).
Did you hit some specific case where this was blocking?